### DISCUSSION TITLE
How to enable DLSS?

### DISCUSSION THREAD

**Discussion by francescofugazzi** [NVlabs/instant-ngp#603] [Q&A]
Is there any way to enable DLSS on the new version?

**Comment by cpkcpkcpk** [comment]
I'm also confused about this. Is DLSS disabled by default?

**Comment by Tom94** [comment]
Yes, it is disabled by default. If you have Vulkan installed (and it is correctly detected by CMake), __instant-ngp__ is built with DLSS support. If you then start the GUI, it'll attempt to initialize Vulkan and NGX -- only if this step succeeds (check the console log) will there be a "DLSS" checkbox in the GUI that you can enable. Cheers!

**Comment by francescofugazzi** [comment]
I installed Vulkan SDK and then I tried again the first cmake command. No mention about Vulkan in the list and it reports also that DLSS can't be enabled as it requires CUDA 11.x and ind i have 11.6

-- Building for: Visual Studio 17 2022
-- The C compiler identification is MSVC 19.31.31107.0
-- The CXX compiler identification is MSVC 19.31.31107.0
-- The CUDA compiler identification is NVIDIA 11.6.124
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.31.31103/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.31.31103/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting CUDA compiler ABI info
-- Detecting CUDA compiler ABI info - done
-- Check for working CUDA compiler: C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v11.6/bin/nvcc.exe - skipped
-- Detecting CUDA compile features
-- Detecting CUDA compile features - done
-- Targeting GPU architectures: 86
CMake Warning at CMakeLists.txt:130 (message):
  CUDA version too old for DLSS support.  Requires CUDA 11.X


-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE
-- Using Win32 for window creation
-- Found OpenMP_C: -openmp (found version "2.0")
-- Found OpenMP_CXX: -openmp (found version "2.0")
-- Found OpenMP: TRUE (found version "2.0")
-- OptiX_INSTALL_DIR value: C:\ProgramData\NVIDIA Corporation\OptiX SDK 7.4.0
-- Found Python: C:/Program Files/WindowsApps/PythonSoftwareFoundation.Python.3.9_3.9.3312.0_x64__qbz5n2kfra8p0/python3.9.exe (found suitable version "3.9.12", minimum required is "3.7") found components: Interpreter Development Development.Module Development.Embed
-- pybind11 v2.7.1
CMake Warning (dev) at C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.22/Modules/CMakeDependentOption.cmake:84 (message):
  Policy CMP0127 is not set: cmake_dependent_option() supports full Condition
  Syntax.  Run "cmake --help-policy CMP0127" for policy details.  Use the
  cmake_policy command to set the policy and suppress this warning.
Call Stack (most recent call first):
  dependencies/pybind11/CMakeLists.txt:98 (cmake_dependent_option)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Performing Test HAS_MSVC_GL_LTCG
-- Performing Test HAS_MSVC_GL_LTCG - Success
-- Configuring done
-- Generating done
-- Build files have been written to: E:/instant-ngp/instant-ngp/build

**Comment by francescofugazzi** [comment]
I tried to write the exact version  in the CMakeLists.txt and still it thinks it's not the correct CUDA version
I removed the IF-ELSE part and it went on finding VULKAN, right now it's compliling

**Comment by francescofugazzi** [comment]
it fails compiling :( 
I also installed NGX SDK but I keep on getting this error when compiling
It doesn't find nvsdk_ngx_vk.h
How can this be solved?

E:\instant-ngp\instant-ngp\src\dlss.cu(36): fatal error C1083: Non Þ possibile aprire il file inclusione: 'nvsdk_ngx_vk.h': N
o such file or directory [E:\instant-ngp\instant-ngp\build\ngp.vcxproj]
  dlss.cu
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\BuildCustomizations\CUDA 11.6.targets(790,9
): error MSB3721: uscita dal comando ""C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.6\bin\nvcc.exe" -gencode=arch=c
ompute_86,code=\"compute_86,compute_86\" -gencode=arch=compute_86,code=\"sm_86,compute_86\" --use-local-env -ccbin "C:\Progra
m Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.31.31103\bin\HostX64\x64" -x cu -rdc=true  -I"E:\instant-ngp\
instant-ngp\dependencies\tiny-cuda-nn\include" -I"E:\instant-ngp\instant-ngp\dependencies\tiny-cuda-nn\dependencies" -IC:\Vul
kanSDK\1.3.211.0\Include -I"E:\instant-ngp\instant-ngp\dependencies\dlss\include" -I"E:\instant-ngp\instant-ngp\dependencies\
glfw\include" -I"E:\instant-ngp\instant-ngp\dependencies\imgui" -I"E:\instant-ngp\instant-ngp\dependencies\gl3w" -I"E:\instan
t-ngp\instant-ngp\dependencies" -I"E:\instant-ngp\instant-ngp\dependencies\eigen" -I"E:\instant-ngp\instant-ngp\dependencies\
filesystem" -I"E:\instant-ngp\instant-ngp\dependencies\nanovdb" -I"C:\ProgramData\NVIDIA Corporation\OptiX SDK 7.4.0\include"
 -I"E:\instant-ngp\instant-ngp\dependencies\tinylogger" -I"E:\instant-ngp\instant-ngp\include" -I"E:\instant-ngp\instant-ngp\
build" -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.6\include"     --keep-dir x64\RelWithDebInfo  -maxrregcount=
0  --machine 64 --compile -cudart static --extended-lambda --expt-relaxed-constexpr -std=c++14 -Xcompiler="/EHsc -Zi -Ob1 -bi
gobj"   -D_WINDOWS -DNDEBUG -DNGP_VULKAN -DGLFW_INCLUDE_VULKAN -DNGP_GUI -DNGP_OPTIX -D"NGP_VERSION=\"1.0dev\"" -DTCNN_MIN_GP
U_ARCH=86 -DTCNN_SHAMPOO -D"CMAKE_INTDIR=\"RelWithDebInfo\"" -D_MBCS -DWIN32 -D_WINDOWS -D_CRT_SECURE_NO_WARNINGS -DNDEBUG -D
NGP_VULKAN -DGLFW_INCLUDE_VULKAN -DNGP_GUI -DNGP_OPTIX -D"NGP_VERSION=\"1.0dev\"" -DTCNN_MIN_GPU_ARCH=86 -DTCNN_SHAMPOO -D"CM
AKE_INTDIR=\"RelWithDebInfo\"" -Xcompiler "/EHsc /W1 /nologo /O2 /FdE:\instant-ngp\instant-ngp\build\RelWithDebInfo\ngp.pdb /
FS /Zi  /MD /GR" -o ngp.dir\RelWithDebInfo\dlss.obj "E:\instant-ngp\instant-ngp\src\dlss.cu"" con codice 2. [E:\instant-ngp\i
nstant-ngp\build\ngp.vcxproj]

**Reply by Tom94** [reply]
These error look to me like you need to update submodules.

NGX is part of the DLSS submodule that was recently introduced, and the CUDA version check was fixed in tiny-cuda-nun about a week ago.

**Reply by francescofugazzi** [reply]
that was it!!!
It worked perfectly. Thank you very much

**Reply by cpkcpkcpk** [reply]
Worked for me too, thanks!

**Reply by TIMESTICKING** [reply]
> These error look to me like you need to update submodules.
> 
> NGX is part of the DLSS submodule that was recently introduced, and the CUDA version check was fixed in tiny-cuda-nun about a week ago.

I'm not sure how to update submodules, can you offer me a link or something abt it? Thx!

**Reply by TIMESTICKING** [reply]
and also my gpu card is 1080Ti, and DLSS doesn't support it. While I'm building cmake, it finds the Vulkan path, but I cant use it, could it be the reason of my 1080ti card

**Reply by compuword** [reply]
Você pode baixar o modulo já compilado com este recurso ativado se o seu cartão suportar: 

 [**RTX 5000 series and other Blackwell cards**](https://github.com/NVlabs/instant-ngp/releases/download/continuous/Instant-NGP-for-RTX-5000.zip)
- [**RTX 3000 & 4000 series, RTX A4000&ndash;A6000**, and other Ampere & Ada cards](https://github.com/NVlabs/instant-ngp/releases/download/continuous/Instant-NGP-for-RTX-3000-and-4000.zip)
- [**RTX 2000 series, Titan RTX, Quadro RTX 4000&ndash;8000**, and other Turing cards](https://github.com/NVlabs/instant-ngp/releases/download/continuous/Instant-NGP-for-RTX-2000.zip)
- [**GTX 1000 series, Titan Xp, Quadro P1000&ndash;P6000**, and other Pascal cards](https://github.com/NVlabs/instant-ngp/releases/download/continuous/Instant-NGP-for-GTX-1000.zip)

caso queira baixar o Vulkan SDK, segue o link do repositório oficial: https://vulkan.lunarg.com/sdk/home
Espero ajudar
